packages/dashboard-hardhat-plugin/test/fixture-projects/hardhat-project-incompatible/artifacts/build-info/798162fd0d2a9ea1d465e324525eeaa7.json

```diff
@@ -0,0 +1 @@
2 SUB PUSH1 0x0 DUP4 ADD MSTORE PUSH2 0x1C2 DUP2 PUSH2 0x186 JUMP JUMPDEST SWAP1 POP SWAP2 SWAP1 POP JUMP INVALID LOG2 PUSH5 0x6970667358 0x22 SLT KECCAK256 0xBC 0xF6 RETURNDATACOPY PUSH15 0x29272F45F7CFDFE5E1C60A172D180B CODESIZE LOG4 SSTORE DUP2 NOT PUSH19 0x9E7137E08604EE64736F6C6343000811003300 ","sourceMap":"66:220:0:-:0;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;135:25;;;:::i;:::-;;;;;;;:::i;:::-;;;;;;;;167:117;;;;;;;;;;;;;:::i;:::-;;:::i;:::-;;135:25;;;;:::o;167:117::-;221:1;208:10;:14;;;;238:39;;;;;;:::i;:::-;;;;;;;;167:117;:::o;7:77:1:-;44:7;73:5;62:16;;7:77;;;:::o;90:118::-;177:24;195:5;177:24;:::i;:::-;172:3;165:37;90:118;;:::o;214:222::-;307:4;345:2;334:9;330:18;322:26;;358:71;426:1;415:9;411:17;402:6;358:71;:::i;:::-;214:222;;;;:::o;523:117::-;632:1;629;622:12;769:122;842:24;860:5;842:24;:::i;:::-;835:5;832:35;822:63;;881:1;878;871:12;822:63;769:122;:::o;897:139::-;943:5;981:6;968:20;959:29;;997:33;1024:5;997:33;:::i;:::-;897:139;;;;:::o;1042:329::-;1101:6;1150:2;1138:9;1129:7;1125:23;1121:32;1118:119;;;1156:79;;:::i;:::-;1118:119;1276:1;1301:53;1346:7;1337:6;1326:9;1322:22;1301:53;:::i;:::-;1291:63;;1247:117;1042:329;;;;:::o;1377:169::-;1461:11;1495:6;1490:3;1483:19;1535:4;1530:3;1526:14;1511:29;;1377:169;;;;:::o;1552:175::-;1692:27;1688:1;1680:6;1676:14;1669:51;1552:175;:::o;1733:366::-;1875:3;1896:67;1960:2;1955:3;1896:67;:::i;:::-;1889:74;;1972:93;2061:3;1972:93;:::i;:::-;2090:2;2085:3;2081:12;2074:19;;1733:366;;;:::o;2105:419::-;2271:4;2309:2;2298:9;2294:18;2286:26;;2358:9;2352:4;2348:20;2344:1;2333:9;2329:17;2322:47;2386:131;2512:4;2386:131;:::i;:::-;2378:139;;2105:419;;;:::o"},"methodIdentifiers":{"set(uint256)":"60fe47b1","storedData()":"2a1afcd9"}},"metadata":"{\"compiler\":{\"version\":\"0.8.17+commit.8df45f5f\"},\"language\":\"Solidity\",\"output\":{\"abi\":[{\"anonymous\":false,\"inputs\":[{\"indexed\":false,\"internalType\":\"string\",\"name\":\"_message\",\"type\":\"string\"}],\"name\":\"StorageSet\",\"type\":\"event\"},{\"inputs\":[{\"internalType\":\"uint256\",\"name\":\"x\",\"type\":\"uint256\"}],\"name\":\"set\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"inputs\":[],\"name\":\"storedData\",\"outputs\":[{\"internalType\":\"uint256\",\"name\":\"\",\"type\":\"uint256\"}],\"stateMutability\":\"view\",\"type\":\"function\"}],\"devdoc\":{\"kind\":\"dev\",\"methods\":{},\"version\":1},\"userdoc\":{\"kind\":\"user\",\"methods\":{},\"version\":1}},\"settings\":{\"compilationTarget\":{\"contracts/SimpleStorage.sol\":\"SimpleStorage\"},\"evmVersion\":\"london\",\"libraries\":{},\"metadata\":{\"bytecodeHash\":\"ipfs\"},\"optimizer\":{\"enabled\":false,\"runs\":200},\"remappings\":[]},\"sources\":{\"contracts/SimpleStorage.sol\":{\"keccak256\":\"0x05045cd5a0727b4860c77a6e26e46c34e2fb0c71ecacc0b0d2ece996c9a7b280\",\"license\":\"MIT\",\"urls\":[\"bzz-raw://3d0ff38592281e6b87cb07abe08af3d6d6e08fa0587e6283c7102c2c2c097701\",\"dweb:/ipfs/QmdaDGp4N3UjGmTPEj4Ap9p3nW7VUTLV5YQy11JsHgTMjx\"]}},\"version\":1}"}}}}}
```

Comment:
Oh it actually (ideally) shouldn't get modified during testing (note the `force: false` [here](https://github.com/trufflesuite/truffle/blob/feat/dashboard-hardhat-plugin/packages/dashboard-hardhat-plugin/test/hardhat.test.ts#L45)) and was consciously left in with an incorrect `_format` entry of `hh-sol-build-info-2` which causes an `IncompatibleHardhatBuildInfoFormatError` to be thrown (although as discussed this was having some problems in CI due to what appears to be an [issue with Chai's](https://www.chaijs.com/api/bdd/#method_instanceof) `instanceOf` assertion when using TS:
```
.instanceof may not always work as expected when using a transpiler such as Babel or TypeScript. In particular, it may produce unexpected results when subclassing built-in object such as Array, Error, and Map
```
That all said, if it's getting modified I'll take a look 🙂

Reply:
Can we just .gitignore the HH artifacts and cache altogether? We can modify the `_format` or whatever we like post-compile by just `fs.writeFile()`-ing one of the build-info files that results from compilation. I don't think they need to be in source control at all?

Reply:
Just been digging into this and the challenge with doing it "post-compile" is that the act of compilation (at least the one in using `test/fixture-projects/hardhat-project-incompatible/hardhat.config.ts`) actually triggers the deployment so it's too late in the lifecycle to attempt to modify the `build-info`.

One option could be to have a separate `hardhat.config.ts` that doesn't import the plugin and that we use to compile first before modifying the `build-info` and re-running it with one that does. That said, after some cursory digging I'm not even sure if it's possible. It's also a little messy.

Hopefully that makes sense 😄 In the meantime I'll do some more digging into options.

Reply:
> ... that the act of compilation actually triggers ...

D'oh, of course!

Another [roughly equally gross] option is to have the tests define their own HH compile plugin that we inject _between_ HH's compilation and this plugin, for the express purpose of modifying those build-info files. But this requires we can tightly control hook plugin ordering and such, and also do we really want a plugin inside the tests for a plugin?

As unideal as it is, I think having a separate `hardhat.config.ts` is better than my idea and also better than including the build-info files in source control.

Reply:
Would be nice if there was a way to tell HH to run without plugins, but I don't see any such mechanism

Reply:
Awesome, thanks (actually spotted it too). First pass [here](https://github.com/trufflesuite/truffle/pull/5927/commits/58a2f0f38d34dda8e2c37689083994615f2563f5#diff-14b74f2ba79d9e8e2f2cee6320711cc35827ca6d2bf09b1ad1b87e10ed4f052b)...it could do with being a bit more DRY but thoughts on approach welcome before I tidy. Also shoutout to GH copilot for the boilerplate 😅